File: core/src/com/broll/gainea/server/core/cards/impl/direct/C_LonelySoldier.java
package com.broll.gainea.server.core.cards.impl.direct;

import com.broll.gainea.net.NT_Abstract_Event;
import com.broll.gainea.server.core.cards.DirectlyPlayedCard;
import com.broll.gainea.server.core.map.Location;
import com.broll.gainea.server.core.objects.Monster;
import com.broll.gainea.server.core.objects.Soldier;
import com.broll.gainea.server.core.objects.buffs.BuffType;
import com.broll.gainea.server.core.objects.buffs.IntBuff;
import com.broll.gainea.server.core.utils.LocationUtils;
import com.broll.gainea.server.core.utils.UnitControl;

public class C_LonelySoldier extends DirectlyPlayedCard {

    public C_LonelySoldier() {
        super(18, "Expedition ins Ungewisse", "Ihr erhaltet einen Soldaten auf einem zufälligen freien Feld");
    }

    @Override
    protected void play() {
        Location location = LocationUtils.getRandomFree(game.getMap().getAllAreas());
        if (location != null) {
            Soldier soldier = owner.getFraction().createSoldier();
            UnitControl.spawn(game, soldier, location);
        }
    }

}

File: server/src/main/com/broll/gainea/server/core/cards/impl/direct/C_LonelySoldier.kt
package com.broll.gainea.server.core.cards.impl.direct

import com.broll.gainea.server.core.cards.DirectlyPlayedCard
import com.broll.gainea.server.core.cards.EffectType.SUMMON
import com.broll.gainea.server.core.utils.UnitControl.spawn
import com.broll.gainea.server.core.utils.getRandomFree

class C_LonelySoldier : DirectlyPlayedCard(
    18,
    SUMMON,
    "Expedition ins Ungewisse",
    "Ihr erhaltet einen Soldaten auf einem zufälligen freien Feld"
) {
    override fun play() {
        val location = game.map.allAreas.getRandomFree()
        if (location != null) {
            val soldier = owner.fraction.createSoldier()
            game.spawn(soldier, location)
        }
    }
}
